src/main/java/uk/gov/hmcts/reform/pcs/ccd/page/createpossessionclaim/ResumeClaim.java

```
@@ -0,0 +1,74 @@
+
+    private final UnsubmittedCaseDataService unsubmittedCaseDataService;
+    private final ModelMapper modelMapper;
+
+    @Override
+    public void addTo(PageBuilder pageBuilder) {
+        pageBuilder
+            .page("resumeClaim", this::midEvent)
+            .pageLabel("Resume claim")
+            .showCondition("hasUnsubmittedCaseData=\"Yes\"")
+            .readonly(PCSCase::getHasUnsubmittedCaseData, NEVER_SHOW)
+            .label("resumeClaim-info", """
+                ---
+                <p class="govuk-body">
+
+                Your previous answers have been saved. You can either:
+
+                <ul class="govuk-list">
+                  <li class="govuk-!-font-size-19">resume your claim using your saved answers</li>
+                  <li class="govuk-!-font-size-19">start your claim again from the beginning</li>
+                </ul>
+
+                If you resume your claim using your saved answers, you must select ‘Continue’ at the bottom of each
+                page until you reach the question you’d like to continue your claim from.
+
+                </p>
+                """)
+            .mandatory(PCSCase::getResumeClaimKeepAnswers);
+
+    }
+
+    private AboutToStartOrSubmitResponse<PCSCase, State> midEvent(CaseDetails<PCSCase, State> details,
+                                                                  CaseDetails<PCSCase, State> detailsBefore) {
+
+        long caseReference = details.getId();
+        PCSCase caseData = details.getData();
+
+        log.debug("Resuming claim - keep existing answers = " + caseData.getResumeClaimKeepAnswers());
+
+        if (caseData.getResumeClaimKeepAnswers() == YesOrNo.YES) {
+            unsubmittedCaseDataService.getUnsubmittedCaseData(caseReference)
+                .ifPresentOrElse(
+                    unsubmittedCaseData -> modelMapper.map(unsubmittedCaseData, caseData),
+                    () -> log.warn("No unsubmitted case data found for case " + caseReference)
```

It it worth throwing an error here instead of a log? If the draft data doesn't exist it just proceeds and they won't know from what I'm seeing here?

---

It's a good point. I had deliberately decided to let the user carry on because they've said Yes to keeping the existing answers, but there are no existing answers so there's nothing to keep.

In practice I think the main way for this to occur would be if someone/something else completed the claim event concurrently between this user being shown the screen to decide whether to keep their answers and them making and submitting their choice.

My main objective was to avoid having an error that prevented the user from doing anything, but on reflection I could throw an error here and the user could just try again and choose "No" to keeping the answers, in order to be able to proceed.

I'll update the code now